### PR TITLE
openssl: add embed-bitcode if tools.apple:enable_bitcode is set

### PR DESCRIPTION
**Pull request by kletoz**
Specify library name and version:  **openssl/1.x.x** and **openssl/3.x.x**.

Add `-fembed-bitcode` to `CFLAGS` if `tools.apple:enable_bitcode` is configured; only for `iOS` and `tvOS`.

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Nekto89** [comment]
Doesn't watchOS also use this flag?

File: recipes/openssl/1.x.x/conanfile.py [hunk 1]
```diff
@@ -757,6 +757,8 @@ def build(self):
                 xcrun = tools.XCRun(self.settings)
                 env_vars["CROSS_SDK"] = os.path.basename(xcrun.sdk_path)
                 env_vars["CROSS_TOP"] = os.path.dirname(os.path.dirname(xcrun.sdk_path))
+                if self.settings.os in ["iOS", "tvOS", "watchOS"] and self.conf.get("tools.apple:enable_bitcode", check_type=bool):
+                    env_vars["CFLAGS"] = "-fembed-bitcode"
```
Comment (SpaceIm) [inline recipes/openssl/1.x.x/conanfile.py:761]: it overrides CFLAGS instead of appending to CFLAGS

File: recipes/openssl/1.x.x/conanfile.py [hunk 2]
```diff
@@ -757,6 +757,9 @@ def build(self):
                 xcrun = tools.XCRun(self.settings)
                 env_vars["CROSS_SDK"] = os.path.basename(xcrun.sdk_path)
                 env_vars["CROSS_TOP"] = os.path.dirname(os.path.dirname(xcrun.sdk_path))
+                if self.settings.os in ["iOS", "tvOS", "watchOS"] and self.conf.get("tools.apple:enable_bitcode", check_type=bool):
+                    cflags = " ".join(self._get_env_build().vars_dict["CFLAGS"])
+                    env_vars["CFLAGS"] = "%s %s" % (cflags, "-fembed-bitcode")
```
Comment (prince-chrismc) [inline recipes/openssl/1.x.x/conanfile.py:762]: 🤔 I'd expect the same as boost

https://github.com/conan-io/conan-center-index/blob/e650a185c988ef064029c41e9fd56e79ad4a01ae/recipes/boost/all/conanfile.py#L1116-L1117

**Comment by kletoz** [comment]
@SpaceIm, @prince-chrismc could you please recheck this PR. It's adding bitcode for Apple builds.

**Pull request closed without merging**